modules/codegen/src/main/scala/com/twilio/guardrail/generators/Http4sServerGenerator.scala

```diff
@@ -217,7 +217,7 @@ object Http4sServerGenerator {
             Target.pure(
               Param(
                 None,
-                Some((q"urlForm.values.get(${arg.argName.toLit}).map(_.map(Json.fromString(_).as[$tpe]))", p"Some(Seq(Right(${Pat.Var(arg.paramName)})))")),
+                Some((q"urlForm.values.get(${arg.argName.toLit}).flatMap(_.toList.headOption).map(Json.fromString(_).as[$tpe])", p"Some(Right(${Pat.Var(arg.paramName)}))")),
```

Comment:
Do we need to force `toList` here? If it's an iterable, can we not just `headOption` without first going to `List`?

Reply:
Agree. Question here would be is a `headOption` is enough? In prev. solution we had a `Option[Seq[Either]]` now it's just an `Option[Either]]`. Maybe it's better to have it like `Seq[Either]`?

Reply:
The semantics are different here, definitely, but the tradeoffs are this:

If there are multiple values supplied with the same name,
- 🚀... do we attempt to parse all of them and fail if any of them are invalid, and only if all of them are valid do we take the first one?
- 😄... do we take the first one and ignore all others, even if that one is invalid but others are valid?
- ❤️... do we take the _last_ one and ignore all others, even if that one is invalid but others are valid?
- 🎉... do we validate first, ignoring errors, then take the first/last?

Vote with emoji!

Reply:
I've took a deep look at `directivesFromParams` that is used here and the first parameter assumes that we have one mandatory fields, so vote with 😄

Reply:
How about a fifth option?  If we are expecting a single value but receive multiple, fail immediately, because clearly the client is doing something wrong and will get unexpected behavior

Reply:
@kelnos do you think it's possible to test? I mean we generate client and server and we assume having a client that accepts certain types. I guess it's not a valid case.

Reply:
Sure.  You wouldn't be able to use the autogenerated client to test (since it would of course conform to the correct behavior), but you could build `HttpRequest`s from scratch and send them to the server impl.

Reply:
@kelnos please check my latest commits I think I've resolved all the issues.